runtime/common/xwalk_resource_delegate.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2016 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_COMMON_XWALK_RESOURCE_DELEGATE_H_
+#define XWALK_RUNTIME_COMMON_XWALK_RESOURCE_DELEGATE_H_
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "base/macros.h"
+#include "base/memory/ref_counted_memory.h"
+#include "ui/base/resource/resource_bundle.h"
+
+namespace base {
+class FilePath;
+}
+
+namespace gfx {
+class Image;
+}
+
+namespace xwalk {
+
+// A singleton resource bundle delegate. Primary purpose is to indicate the
```

Comment:
The explanation here needs to be much more meaningful. Put yourself in the shoes of someone who will look at this code 2 years from now: it is still unclear why the delegate is needed (isn't there code in `ui` doing that automatically? what's wrong with locales in `locales/`?)